src/sentry/rules/processing/delayed_processing.py

```diff
@@ -1,36 +1,273 @@
+            # previous interval to compare against the current interval
+            if condition_data:
+                comparison_type = condition_data.get("comparisonType", ComparisonType.COUNT)
+                if comparison_type == ComparisonType.PERCENT:
+                    comparison_interval = condition_inst.intervals[unique_condition.interval][1]
+                    comparison_end = end - comparison_interval
+                    start = comparison_end - duration
+                    comparison_results = condition_inst.batch_query(
+                        group_ids=group_ids,
+                        start=start.replace(tzinfo=UTC),
+                        end=comparison_end.replace(tzinfo=UTC),
+                        environment_id=unique_condition.environment_id,
+                    )
+
+                    results = {
+                        group_id: percent_increase(results[group_id], comparison_results[group_id])
+                        for group_id in group_ids
+                    }
+
+        condition_group_results[unique_condition] = results
+    return condition_group_results
+
+
+def get_rules_to_fire(
+    condition_group_results: dict[UniqueCondition, dict[int, int]],
+    rule_to_slow_conditions: DefaultDict[Rule, list[dict[str, Any]]],
+    rules_to_groups: DefaultDict[int, set[int]],
+) -> DefaultDict[Rule, list[int]]:
+    condition_to_results = {c.cls_id: results for c, results in condition_group_results.items()}
+    rules_to_fire = defaultdict(set)
+    for alert_rule, slow_conditions in rule_to_slow_conditions.items():
+        for slow_condition in slow_conditions:
+            condition_id = slow_condition.get("id")
+            target_value = int(slow_condition.get("value"))
+            results = condition_to_results[condition_id]
+            for group_id in rules_to_groups[alert_rule.id]:
+                if results[group_id] > target_value:
+                    rules_to_fire[alert_rule].add(group_id)
+    return rules_to_fire
+
+
 @redis_buffer_registry.add_handler(BufferHookEvent.FLUSH)
 def process_delayed_alert_conditions(buffer: RedisBuffer) -> None:
     with metrics.timer("delayed_processing.process_all_conditions.duration"):
         project_ids = buffer.get_set(PROJECT_ID_BUFFER_LIST_KEY)
 
         for project in RangeQuerySetWrapper(Project.objects.filter(id__in=project_ids)):
-            rulegroup_event_mapping = buffer.get_hash(
-                model=Project, field={"project_id": project.id}
-            )
-
             with metrics.timer("delayed_processing.process_project.duration"):
-                safe_execute(
-                    apply_delayed,
-                    project,
-                    rulegroup_event_mapping,
-                    _with_transaction=False,
-                )
+                apply_delayed.delay(project=project, buffer=buffer)
+
+
+@instrumented_task(
+    name="sentry.delayed_processing.tasks.apply_delayed",
+    queue="dynamicsampling",
```

Comment:
Is it ok to use this queue? If not, how can we choose/create a new/different one?

Reply:
I'm not sure the impact of this 😬 -- cc @wedamija

Reply:
By specifying no queue do you mean just not pass it for now?

Reply:
Yep, you can just pass no queue and it'll go to default